getter.go

```diff
@@ -144,6 +144,7 @@ func (g *getter) queueFile(url *url.URL) (http.Header, error) {
 	// resp could be nil, depending on the error.
 	if err != nil {
 		g.chunkWg.Done()
+		defer resp.Body.Close()
```

Comment:
should we just move this defer to be after the `g.retryRequest()` ;  it is the defer action that is related to that call.   then we don't have to independently call it in each logic case below

need to check the contract of retryRequest and if we are guaranteed to have a non-nil resp and resp.Body to ensure resp.Body.Close() is always acceptable to blindly call